components/layout_thread_2020/lib.rs

```diff
@@ -1012,7 +1012,8 @@ impl LayoutThread {
             RecalcStyle::pre_traverse(dirty_root, shared)
         };
 
-        let rayon_pool = STYLE_THREAD_POOL.pool();
+        let rayon_pool = STYLE_THREAD_POOL.lock().unwrap();
```

Comment:
Instead of locking and unlocking the pool above in `build_layout_context`, it might be better to only grab the mutex above and pass a new `use_rayon: bool` argument to `build_layout_context`.